### DISCUSSION TITLE
Recommended way to access and alter collection data in a module

### DISCUSSION THREAD

**Discussion by heitordelima** [directus/directus#7336] [Archive]
I'm working on a module that should show a table with a collection and its relationship data and should be able to update a field for each item. What's the best way to access these collection data from a module?

**Comment by nickrum** [comment]
The provided `system` field provides access to the collections store and the API to be able to read and update collection data. There is an example explaining this in the [docs](https://docs.directus.io/guides/modules/#accessing-the-api-from-within-your-extension).

**Reply by heitordelima** [reply]
So I have to write all logic to watch and update values. I thought that Directus provided some service or "magic" stuff kinda like what we do with the custom interfaces. Thank you for the information.
